ui/packages/catalog-ui-search/src/main/webapp/component/query-advanced/query-custom/query-custom.view.js

```diff
@@ -65,6 +72,56 @@ define([
             this.model.set({
                 cql: filter
             });
+        },
+        checkIfDefaultSearchForm: function() {
+            var storedTemplate = user.getQuerySettings().toJSON();
+            var currentTemplate = this.options.model;
+
+            if (storedTemplate['defaultTemplate'] != undefined &&
+                storedTemplate['defaultTemplate'].model.title == currentTemplate.get('title')) {
```

Comment:
`!== undefined` and `storedTemplate['defaultTemplate'].model.title === currentTemplate.get('title')`?